build/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM golang:1.5
+
+MAINTAINER Peter Mattis <peter@cockroachlabs.com>
+
+RUN apt-get update -y && \
+ apt-get dist-upgrade -y && \
+ apt-get install --no-install-recommends --auto-remove -y git build-essential file npm nodejs && \
+ apt-get clean autoclean && \
+ apt-get autoremove -y && \
+ rm -rf /tmp/* && \
+ ln -s /usr/bin/nodejs /usr/bin/node
+RUN go get golang.org/x/tools/cmd/vet
+
+COPY npm-shrinkwrap.json package.json ./
+RUN npm install
+RUN rm npm-shrinkwrap.json package.json
```

Comment:
Each action creates a new docker layer which is why you often see these long command lines separated by `&&`. Probably worthwhile to do:

```
  RUN npm install && \
    rm npm-shrinkwrap.json package.json
```

Might even be worthwhile to combine with the `RUN` above.